app/contexts/ThemeProvider.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { createContext, useContext, useEffect, useState } from 'react'
+import { ColorSchemeName, useColorScheme } from 'react-native'
+import { Logging } from '../api'
+import { ThemePersistence } from '../api/wallet/theme_storage'
+
+interface ThemeLoader {
+  theme: NonNullable<ColorSchemeName>
+  isThemeLoaded: boolean
+}
+
+export function useTheme (): ThemeLoader {
+  const colorScheme = useColorScheme()
+  const [isThemeLoaded, setIsThemeLoaded] = useState<boolean>(false)
+  const [theme, setTheme] = useState<NonNullable<ColorSchemeName>>('light')
+
+  useEffect(() => {
+    ThemePersistence.get().then((t) => {
+      let currentTheme: NonNullable<ColorSchemeName> = 'light'
+      if (t !== null && t !== undefined) {
+        currentTheme = t as NonNullable<ColorSchemeName>
+      } else if (colorScheme !== null && colorScheme !== undefined) {
+        currentTheme = colorScheme
+      }
+      setTheme(currentTheme)
+    }).catch((err) => Logging.error(err)).finally(() => setIsThemeLoaded(true))
+  }, [])
+
+  return {
+    isThemeLoaded,
+    theme
+  }
+}
+
+const light: Record<string, string> = {
+  'body-bg': 'bg-gray-100',
+  'row-bg': 'bg-white',
+  'row-border': 'border-b border-gray-200',
+  'title-text': 'text-gray-500',
+  'body-text': 'text-black',
+  'subtitle-text': 'text-gray-600',
+  'text-primary': 'text-primary',
+  'light-text': 'text-gray-300'
+}
+
+const dark: Record<string, string> = {
+  'body-bg': 'bg-dark',
+  'row-bg': 'bg-darksurface',
+  'row-border': 'border-b border-dark',
+  'title-text': 'text-white text-opacity-90',
+  'body-text': 'text-white text-opacity-90',
+  'subtitle-text': 'text-white text-opacity-70',
+  'text-primary': 'text-darkprimary',
+  'light-text': 'text-gray-300'
+}
+
+interface Theme {
+  theme: NonNullable<ColorSchemeName>
+  setTheme: (theme: NonNullable<ColorSchemeName>) => void
+  getThemeClass: (type: string) => string
```

Comment:
```suggestion
  tailwind: function
```

allowing: 
```ts
const { tailwind } = useTheme()
```

We should do this instead, let the `Theme` Context automatically convert without having to use `getThemeClass`. 

By introducing `getThemeClass` you introduce the CSS Zen Garden concept which is an anti pattern of utility first css - tailwind. This sort of approach is hard to manage and scale leading up to a concentration of semantic to maintain. Who maintain this semantics, how does it evolve, does it break another screen? Leading up to the problems which leds to the eventual creation of utility first css.

@aentan shared this a few months back:
> https://adamwathan.me/css-utility-classes-and-separation-of-concerns/
> IMO this is the only sure way to really get UI consistency and polish. In my 18 years writing FE code, I’ve done it all. CSS Zen Garden semantic CSS, BEM, etc. I’ve arrived at the same conclusion as the above author.

I share the same opinion as @aentan